app/controllers/shared/api.php

```diff
@@ -551,6 +544,22 @@
                 ->setParam('project.{scope}.network.outbound', $response->getSize())
                 ->submit();
         }
+
+        /**
+         * Update user last activity
+         */
+        if (!$user->isEmpty()) {
+            $accessedAt = $user->getAttribute('accessedAt', '');
+            if (DateTime::formatTz(DateTime::addSeconds(new \DateTime(), -APP_USER_ACCCESS)) > $accessedAt) {
+                $user->setAttribute('accessedAt', DateTime::now());
+
+                if (APP_MODE_ADMIN !== $mode) {
+                    $dbForProject->updateDocument('users', $user->getId(), $user);
+                } else {
+                    $dbForConsole->updateDocument('users', $user->getId(), $user);
```

Comment:
These database updates are better handled in a worker. We are adding avoidable latency to every authenticated request